### PR TITLE
`azurerm_redis_cache`: support for Patch Schedules

### PR DESCRIPTION
**Pull request by tombuildsstuff**
Adding support for Patch Schedules for Premium SKU's

### CODE REVIEW

**Comment by tombuildsstuff** [comment]
Tests pass:

screen shot 2017-11-09 at 12 27 19

File: azurerm/resource_arm_redis_cache.go [hunk 1]
```diff
@@ -218,6 +251,15 @@ func resourceArmRedisCacheCreate(d *schema.ResourceData, meta interface{}) error
 
 	d.SetId(*read.ID)
 
+	// TODO: this _may_ need an `if premium` check
```
Comment (mbfrahry) [inline azurerm/resource_arm_redis_cache.go]: Is this TODO for later or do we want to get it looked at now?

File: azurerm/resource_arm_redis_cache.go [hunk 2]
```diff
@@ -288,6 +330,25 @@ func resourceArmRedisCacheUpdate(d *schema.ResourceData, meta interface{}) error
...
```
Comment (mbfrahry) [inline azurerm/resource_arm_redis_cache.go]: Same as above

**Review by mbfrahry — APPROVED** [review]
LGTM. Just a couple questions

File: azurerm/resource_arm_redis_cache.go (hunk 1 above)
Comment (tombuildsstuff) [reply azurerm/resource_arm_redis_cache.go]: good spot, I'd meant to remove that - I'll do that now..

File: azurerm/resource_arm_redis_cache.go (hunk 2 above)
Comment (tombuildsstuff) [reply azurerm/resource_arm_redis_cache.go]: +1 here

File: azurerm/resource_arm_redis_cache.go (hunk 1 above)
Comment (tombuildsstuff) [reply azurerm/resource_arm_redis_cache.go]: fixed.

**Pull request merged**